src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -382,6 +414,104 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
 
+        private async Task<BuildPath> GetLongestBuildPath(MaestroBuild build, List<string> visitedRepositories, Build[] builds, AzureDevOpsTimelineOptions options)
+        {
+            int buildId = 0;
+            double buildTime = 0;
+            string repository = "";
+
+            if (build == null)
+            {
+                return new BuildPath(buildId, repository, buildTime);
+            }
+
+            // Lookup Kusto build. First check builds, then check the Kusto database.
+            Build kustoBuild = builds.FirstOrDefault(b => b.Id == build.AzureDevOpsBuildId);
+
+            if (kustoBuild == null)
+            {
+                // Build is not in the builds list, so we need to look this build up in Kusto
+                try
+                {
+                    using (ICslQueryProvider query =
+                        KustoClientFactory.CreateCslQueryProvider(options.KustoQueryConnectionString))
+                    using (IDataReader result = await query.ExecuteQueryAsync(
+                        options.KustoDatabase,
+                        // This isn't use controlled, so I'm not worried about the Kusto injection
+                        $"TimelineBuilds | where BuildId == '{build.AzureDevOpsBuildId}' | project BuildId, Repository, StartTime, FinishTime",
```

Comment:
This isn't going to be reliable.  There is a huge lag in Kusto, so you aren't going to find the data you expect to find... this will flutuate wildly.